tensorflow_datasets/core/dataset_builder.py

```diff
@@ -284,7 +284,17 @@ def _checksums_path(cls) -> ReadOnlyPath:
     # Used:
     # * To load the checksums (in url_infos)
     # * To save the checksums (in DownloadManager)
-    return cls.code_path.parent / "checksums.tsv"
+    #Checking whether the dataset follow the folder structure
+    path = str(cls.code_path.parent)
+    if path.split('\\')[-1] == cls.name:
+      return cls.code_path.parent / "checksums.tsv"
+    else:
+      #Record checksums in the common folder for older structured datasets
+      chksm_path = utils.gpath.WindowsGPath(
+        'f:/tfds/datasets/tensorflow_datasets/url_checksums/' +
+        cls.name +
+        '.txt')
+      return chksm_path
```

Comment:
- Don't hardcode any value
- Code should be independent of OS

Try rewriting as:
```
utils.tfds_path() / 'url_checksums' / (cls.name + '.txt')
```